### PR TITLE
Calling Sentry.init and specifying contextTags now has an effect on the Log4j SentryAppender

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

No longer used the `contextTags` property in `SentryAppender`, instead get via Options from HubAdapter.

## :bulb: Motivation and Context

Fix https://github.com/getsentry/sentry-java/issues/2043 for Log4j

## :green_heart: How did you test it?
Demo code

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java [hunk 1]
```diff
@@ -187,6 +187,7 @@ public void append(final @NotNull LogEvent eventObject) {
         CollectionUtils.filterMapEntries(
             loggingEvent.getContextData().toMap(), entry -> entry.getValue() != null);
     if (!contextData.isEmpty()) {
+      List<String> contextTags = HubAdapter.getInstance().getOptions().getContextTags();
```
Comment (romtsn) [inline sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java:190]:
```suggestion
      List<String> contextTags = hub.getOptions().getContextTags();
```
we actually have a `hub` instance already as a class member, can we use that one instead?

**Review by romtsn — APPROVED** [review]
brrrr, this feels wrong again, having to maintain 2 different properties for the same purpose 🤣 

Would it be possible to set `contextTags` directly on the `hub` in the SentryAppender constructor? This way we could eliminate the field `contextTags` and wouldn't need to set it in `start`. 

Although this could be a problem if the appender was initialized but never started, we still set the contextTags on the hub, don't know if that can be ever the case

**Comment by adinauer** [comment]
> brrrr, this feels wrong again, having to maintain 2 different properties for the same purpose 🤣
> 
> Would it be possible to set `contextTags` directly on the `hub` in the SentryAppender constructor? This way we could eliminate the field `contextTags` and wouldn't need to set it in `start`.
> 
> Although this could be a problem if the appender was initialized but never started, we still set the contextTags on the hub, don't know if that can be ever the case

I've added something to #2053 , should we deal with this at a later time?

**Comment by romtsn** [comment]
> > brrrr, this feels wrong again, having to maintain 2 different properties for the same purpose 🤣
> > Would it be possible to set `contextTags` directly on the `hub` in the SentryAppender constructor? This way we could eliminate the field `contextTags` and wouldn't need to set it in `start`.
> > Although this could be a problem if the appender was initialized but never started, we still set the contextTags on the hub, don't know if that can be ever the case
> 
> I've added something to #2053 , should we deal with this at a later time?

Yeah sure, was just a quick suggestion, but we can do it later

**Comment by adinauer** [comment]
> > > brrrr, this feels wrong again, having to maintain 2 different properties for the same purpose 🤣
> > > Would it be possible to set `contextTags` directly on the `hub` in the SentryAppender constructor? This way we could eliminate the field `contextTags` and wouldn't need to set it in `start`.
> > > Although this could be a problem if the appender was initialized but never started, we still set the contextTags on the hub, don't know if that can be ever the case
> > 
> > 
> > I've added something to #2053 , should we deal with this at a later time?
> 
> Yeah sure, was just a quick suggestion, but we can do it later

I like the suggestion but I'm not sure about the consequenes so I don't wanna rush it. The quick fix should suffice for now I hope.

File: sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java [hunk 2]
```diff
@@ -187,6 +187,7 @@ public void append(final @NotNull LogEvent eventObject) {
         CollectionUtils.filterMapEntries(
             loggingEvent.getContextData().toMap(), entry -> entry.getValue() != null);
     if (!contextData.isEmpty()) {
+      List<String> contextTags = hub.getOptions().getContextTags();
```
Comment (marandaneto) [inline sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java:190]:
```suggestion
      final List<String> contextTags = hub.getOptions().getContextTags();
```

**Comment by maciejwalkowiak** [comment]
I believe it should be put on hold until https://github.com/getsentry/sentry-java/pull/2052#pullrequestreview-978110817 is resolved, so that we have ideally one consistent approach between logging frameworks.

**Pull request merged**